sdk/core/Azure.Core/src/Http/TemporalRequestOptions.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Http
+{
+    public class TemporalRequestOptions : ConditionalRequestOptions, IEquatable<TemporalRequestOptions>
```

Comment:
We don't use `Temporal` [very often](https://docs.microsoft.com/en-us/dotnet/api/?term=Temporal) and I think we might want the `Conditional` still in there somewhere.

`ModifiedConditionalRequestOptions`?  It's too long and kind of weird, but I can't think of a good name for this type.

Reply:
To do it out fully, it would be, ModificationDateConditionalRequestOptions.  How do you feel about just DateConditionalRequestOptions?  All the precondition headers are talking about modification.

Reply:
I was coming at it from the base type having all the "Match" headers and this type having all the "ModifiedSince" headers, but I like your proposed name.